web-api/src/Kerbero.Domain/SmartLockKeys/Interactors/OpenSmartLockWithKeyInteractor.cs

```diff
@@ -0,0 +1,96 @@
+using FluentResults;
+using Kerbero.Domain.Common.Errors;
+using Kerbero.Domain.Common.Models;
+using Kerbero.Domain.NukiCredentials.Repositories;
+using Kerbero.Domain.SmartLockKeys.Errors;
+using Kerbero.Domain.SmartLockKeys.Interfaces;
+using Kerbero.Domain.SmartLockKeys.Repositories;
+using Kerbero.Domain.SmartLocks.Interfaces;
+
+namespace Kerbero.Domain.SmartLockKeys.Interactors;
+
+public class OpenSmartLockWithKeyInteractor : IOpenSmartLockWithKeyInteractor
+{
+	private readonly ISmartLockKeyRepository _smartLockKeyRepository;
+	private readonly INukiCredentialRepository _nukiCredentialRepository;
+	private readonly IOpenSmartLockInteractor _openSmartLockInteractor;
+
+	public OpenSmartLockWithKeyInteractor(
+		ISmartLockKeyRepository smartLockKeyRepository,
+		INukiCredentialRepository nukiCredentialRepository,
+		IOpenSmartLockInteractor openSmartLockInteractor
+		)
+	{
+		_smartLockKeyRepository = smartLockKeyRepository;
+		_nukiCredentialRepository = nukiCredentialRepository;
+		_openSmartLockInteractor = openSmartLockInteractor;
+	}
+
+	public async Task<Result> Handle(Guid smartLockKeyId, string smartLockKeyPassword)
+	{
+		var smartLockKeyResult = await _smartLockKeyRepository.GetById(smartLockKeyId);
+		if (smartLockKeyResult.IsFailed)
+		{
+			return Result.Fail(smartLockKeyResult.Errors);
+		}
+
+		var smartLockKey = smartLockKeyResult.Value;
+		var validationResult = _ValidateSmartLockKey(smartLockKey.Password, smartLockKeyPassword, smartLockKey.ExpiryDate);
+		if (validationResult.IsFailed)
+		{
+			return Result.Fail(validationResult.Errors);
+		}
+		
+		// Retrieve information to call open smartlock interactor
+		var smartLockProvider = SmartLockProvider.TryParse(smartLockKey.SmartLockProvider);
+		if (smartLockProvider != SmartLockProvider.Nuki)
+		{
+			return Result.Fail(new UnsupportedSmartLockProviderError());
+		}
```

Comment:
Maybe we can do this earlier? (no point in retrieving / checking the key if we cannot open the smartlock)